project/users/models.py

```diff
@@ -10,6 +10,5 @@ class User(AbstractUser):
     # around the globe.
     name = CharField(_("Name of User"), blank=True, max_length=255)
 
-    @property
```

Comment:
Why did we need to remove this decorator, out of curiosity?

Reply:
This is explained in the commit message for 03770f4

To work through a more concrete example of the problem this causes:

property:
* Declare `get_absolute_url()` as a property
* Go to http://127.0.0.1:8000/admin/users/user/
* Click on any user
* Click the "view on site" link in the top-right
* It will throw `'str' object is not callable`: you can't invoke a string.

method:
* Now switch it to a method
* Click the "view on site" link in the top-right - it will issue a redirect.


Side note: It doesn't issue a redirect to a _useful_ URL because of https://github.com/codebuddies/backend/blob/2683d9a3d9fb3e51287712ae5ce184c178242206/project/core/contrib/sites/migrations/0003_set_site_domain_and_name.py#L16 but lets deal with that as another issue. I'm trying to avoid scope-creep too much in this PR and just focus on the tests. The reason I fixed the signature in this PR which addresses testing (rather than treating it as a seperate bug) is because it was causing a test to fail. Its just that nobody has noticed because you weren't running all the tests.

Reply:
Thanks for the explanation here.

Reply:
On closer inspection, we should remove the redirect method altogether, & also remove the `[view on site] ` button from the Admin.  (see this for how:  https://stackoverflow.com/questions/8662359/django-remove-view-on-site-button-in-the-admin-user-change-form). I think this decorator + method was either left from Coookiecutter or cargo-culted in by myself when piecing together userauth.  Since we are mostly building an API, the redirect for viewing user detail/user profile on the site won't  _ever_ work -- unless we point it at a react URL.  If we do provide a user profile or user detail, it will be as an endpoint, and deliver JSON -- not an HTML template/form.

Reply:
I'll open an issue for that (and probably chuck it into the PR for our testing re-write)...unless there are any objections??

Reply:
The 'view on site' button is probably not that helpful for an API project, but I picked it as an example of an internal call that expects `get_absolute_url` to be a method which exists at the moment, but I think there are other internal calls in django (and possibly DRF) that expect that signature.

If you reeeally want a property version, we could also declare

```py
@property
def absolute_url(self):
    return self.get_absolute_url()
```

> and probably chuck it into the PR for our testing re-write

It can be done as its own PR - should be no dependency/merge conflict there